### ISSUE TITLE
Possible bug: IPs from serialized pop don't see to match what script did

### ISSUE THREAD

**Issue by jonathanhhb** [jonathanhhb/covid_emod_hintschoolclosure#24]
This is one of those "this bug can't be real, I must be doing it wrong" things, but it's time to capture it.

I'm using change_ser_pop.py to assign both Primary and Household IDs to individuals. I'm reporting that out through simple print statements.

Then I feed that .dtk file to the DTK and throw in some print statements to show what IP each individual has to start with. I expect these to match what was done in the prior script.

They don't. At all. Apparently.

I first noticed this because I was trying to test "School Closure" and "WFH" by distributing interventions to everyone in a certain Primary or HH (set actually). It was quickly apparent that the individuals receiving the intervention did not match what I was expecting based on the IP assignments from the script.

**Comment by jonathanhhb** [comment]
A good sanity test -- which I haven't done yet -- might be to just do 10 (or 100) assignments, 1-to-1, such that individual 0 gets assigned Primary:0, individual 1 gets assigned Primary:1, and so on, and then confirm that that's what the DTK gets initialized to. Maybe compare Generic and Environmental.

**Comment by jonathanhhb** [comment]
From some additional testing it seems that during the first 1 or 2 timesteps, individuals don't even have any IPs. ???

**Comment by jonathanhhb** [comment]
I just did a little test where everyone is supposed to have the same IP value (integer) as their id:

Individual 2 updated with IPs: Household:0,Primary:0
Individual 3 updated with IPs: Household:1,Primary:1
Individual 4 updated with IPs: Household:2,Primary:2
Individual 5 updated with IPs: Household:3,Primary:3
Individual 6 updated with IPs: Household:4,Primary:4

**Comment by YeChen-IDM** [comment]
I tested the .dtk file with a simple environmental_sim, the IP information is correct in the .dtk file.

**Comment by jonathanhhb** [comment]
Interesting. In my testing, Individual 0 and Individual 1 have "gone missing". Individual 2 seems to have taken Individual 0's place, and so on. ?

**Comment by YeChen-IDM** [comment]
> Interesting. In my testing, Individual 0 and Individual 1 have "gone missing". Individual 2 seems to have taken Individual 0's place, and so on. ?

My IP is string. But I do noticed that the individual id starts at 2 in my simulation. There is no individual 0 and 1.

**Comment by YeChen-IDM** [comment]
see this ticket for the id starts from 2 issue: https://github.com/InstituteforDiseaseModeling/DtkTrunk/issues/3475

**Comment by jonathanhhb** [comment]
Is it possible there is an issue with the python Serialization class I'm using?

https://github.com/jonathanhhb/covid_emod_hintschoolclosure/commit/c4c157431305ee60f05d7c91b467c9cf195c4c04#diff-59059b149f76318254fef38c0ea6cdfaL21

**Comment by jonathanhhb** [comment]
It's not the python serialization lib. Inspecting the my_sp_file.dtk file I see:

Individual 0 has properties Primary:0.
Individual 0 has properties Household:0.
Individual 1 has properties Primary:1.
Individual 1 has properties Household:1.
Individual 2 has properties Primary:2.
Individual 2 has properties Household:2.
etc.

**Comment by jonathanhhb** [comment]
> see this ticket for the id starts from 2 issue: [InstituteforDiseaseModeling/DtkTrunk#3475](https://github.com/InstituteforDiseaseModeling/DtkTrunk/issues/3475)

Oh wow. Yes, that's exactly what I'm seeing.

**Comment by YeChen-IDM** [comment]
> > see this ticket for the id starts from 2 issue: [InstituteforDiseaseModeling/DtkTrunk#3475](https://github.com/InstituteforDiseaseModeling/DtkTrunk/issues/3475)
> 
> Oh wow. Yes, that's exactly what I'm seeing.
> 
> 3475 doesn't say anything about serialization, though, correct?

Correct, it's not related to serialization. It's a general issue.

**Comment by jonathanhhb** [comment]
Individual 1000 updated with IPs: Household:998,Primary:998
Individual 1001 updated with IPs: Household:999,Primary:999

I think I have 1000 individuals, but their ids go from 2 to 1001, instead of from 0 to 999 per serialization.

**Comment by jonathanhhb** [comment]
It's possible that if I take the as-built "configuration, and increment all my individual ids by 2, everything "checks out".

**Comment by jonathanhhb** [comment]
Indeed, if I add 2 to all the individual ids in my primary and household configurations, the DTK output looks fine!!!
